File: fastify_server/src/endpoint_auth_compute.mjs
/**
 * Endpoint Auth Compute Registry.
 *
 * This file is the entry point for registering custom compute functions for
 * authentication token resolution.
 *
 * Usage:
 *   import { ComputeRegistry } from '@internal/fetch-auth-config';
 *
 *   ComputeRegistry.registerStartup('my_custom_provider', async () => {
 *     // Compute token at startup
 *     return 'my-startup-token';
 *   });
 *
 *   ComputeRegistry.registerRequest('my_request_provider', async (request) => {
 *     // Compute token per request
 *     return request.headers['x-custom-token'];
 *   });
 */

import { ComputeRegistry } from '@internal/fetch-auth-config';

// Add your custom compute functions here

File: fastapi_server/app/endpoint_auth_compute.py
"""
Endpoint Auth Compute Registry.

This file is the entry point for registering custom compute functions for
authentication token resolution.

Usage:
    from fetch_auth_config import register_startup, register_request
    from fastapi import Request

    @register_startup("my_custom_provider")
    async def resolve_my_provider_startup() -> str:
        # Compute token at startup (e.g. fetch from secret manager)
        return "my-startup-token"

    @register_request("my_request_provider")
    async def resolve_my_provider_request(request: Request) -> str:
        # Compute token per request (e.g. extract from header)
        return request.headers.get("X-Custom-Token")
"""

from fetch_auth_config import register_startup, register_request
# from fastapi import Request  # Uncomment if using request resolvers
